src/katsdpcontroller/generator.py

```diff
@@ -208,6 +208,59 @@ def aggregate_compute(self) -> Tuple[Sensor.Status, int]:
         return (status, self._total)
 
 
+class SyncSensor(SimpleAggregateSensor[bool]):
+    """Aggregate which tracks whether its children are synchronised.
+
+    In this case,
+    - a 'synchronised' child has a reading of (True, NOMINAL), and
+    - an 'unsynchronised' child has a reading of (False, ERROR).
+    """
+
+    def __init__(
+        self, target: SensorSet, sensor_type: Type[bool],
+        name: str, description: str, units: str = "",
+        *,
+        auto_strategy: Optional["SensorSampler.Strategy"] = None,
+        auto_strategy_parameters: Iterable[Any] = (),
+        name_regex: re.Pattern,
+        children: int
+    ) -> None:
+        self.target = target
```

Comment:
This line isn't necessary, just pass `target` through to the `super()` method. It does the assignment (which may even just clobber this one? I'm not too sure.)

Reply:
Yeah I'm not sure why I did that. Maybe it was initially a reminder to myself regarding what I have access to. Updating accordingly.